npeps/npep-133-fqdn-egress-selector.md

```diff
@@ -0,0 +1,438 @@
+   *  Pods should respect the TTL of DNS records they receive. Trying to
+      establish new connection using DNS records that are expired is not
+      guaranteed to work.
+   *  When the TTL for a DNS record expires, the implementor should stop
+      allowing new connections to that IP. Existing connection will still be
+      allowed (that's consistent with NetworkPolicy behavior on long-running
+      connections). 
+1. Targeting in-cluster endpoints with FQDN selector:
+   *  ✅︎ Supported:
+      *  Selecting Pods using their [generated DNS
+         record](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods)
+         (for example `pod-ip-address.my-namespace.pod.cluster.local`). This is
+         analogous to selecting the Pod by its IP address using the Network
+         selector.
+      * Headless Services can be selected using their [generated DNS
+        record](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services)
+        because the generated DNS records contain a list of all the Pod IPs that
+        back the service.
+   *  ❌ Not Supported:
+      *  ClusterIP Services can not be selected using their [generated DNS
+         record](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services)
+         (for example `my-svc.my-namespace.svc.cluster.local`). This is
+         consistent with the behavior when selecting the Service VIP using the
+         Network selector.
+      *  ExternalName Services return a `CNAME` record. See the entry below
+         about CNAME support.
+1. If the specified domain in a FQDN selector resolves to a [CNAME
+   record](#cname-records) the behavior of the implementor depends on the
+   returned response.
+
+   If the upstream resolver used [CNAME chasing](#cname-chasing) to fully
+   resolve the domain to a A/AAAA record and returns the resulting chain, the
+   implementor can use this information to allow traffic to the specified IPs.
+   However the implementor does not need to perform their own CNAME chasing or
+   to understand resolutions across multiple DNS requests.
+
+   For example, if the FQDN selector is allowing traffic to `www.kubernetes.io`:
+     *  If a DNS query to the upstream resolver returns *a single response* with
+        the following records:
+        ```
+        www.kubernetes.io  --  CNAME to kubernetes.io
+        kubernetes.io      --  A     to 1.2.3.4
+        ```
+        The implementor can use this response to allow traffic to `1.2.3.4`
+     *  If DNS query only responds with a CNAME record, the resolver is not
+        required to allow traffic even if subsequent requests resolve the full
+        chain:
+        ```
+        # REQUEST 1
+
+        www.kubernetes.io  --  CNAME to kubernetes.io
+        
+        # REQUEST 2
+
+        kubernetes.io      --  A     to 1.2.3.4
+        ```
+        The implementer can still deny traffic to `1.2.3.4` because no single
```

Comment:
Thanks for the clarification. In our implementation I don't think we really looked into this and always assumed cname chasing